multimodal/src/autogluon/multimodal/predictor.py

```diff
@@ -2004,6 +2005,142 @@ def save(self, path: str, standalone: Optional[bool] = True):
+            )
+        elif not batch:
+            if not default_batch:
+                raise ValueError(
+                    f"Do not have default batch to trace for current pipneline {self._pipeline}. Please provide raw data or processed batch as input."
+                )
+
+        torch.onnx.export(
+            model,
+            batch,
+            onnx_path,
+            opset_version=opset_version,
+            verbose=verbose,
+            input_names=valid_input,
+            dynamic_axes=dynamic_axes,
+        )
+
+    def get_processed_batch(
+        self,
+        data: pd.DataFrame,
+        valid_input: Optional[List] = None,
+        onnx_tracing: bool = False,
+        batch_size: int = None,
+        to_numpy: bool = True,
+    ):
+        """
+        Get the processed batch of raw data given.
+
+        Parameters
+        ----------
+        data
+            The raw data to process
+        valid_input
+            Used to filter valid data. No filter happens if it is empty.
+        onnx_tracing
+            If the output is used for onnx tracing.
+        batch_size
+            The batch_size of output batch.
+            If onnx_tracing, it will only output one mini-batch, and all int tensor values will be converted to long.
+        to_numpy
+            Output numpy array if True. Only valid if not onnx_tracing.
+
+        Returns
+        -------
+        Tensor or numpy array.
+        The output processed batch could be used for export/evaluate deployed model.
+        May also be helpful to speed up small batch inference.
+        """
+        # TODO: add support for data = dict or list
+        if onnx_tracing:
+            if batch_size:
+                data = data[:batch_size]
+            else:
+                data = data[:2]
+        data, df_preprocessor, data_processors = self._on_predict_start(
+            config=self._config,
+            data=data,
+            requires_label=False,
+        )
+        modality_features, sample_num = apply_df_preprocessor(
+            data=data,
+            df_preprocessor=df_preprocessor,
+            modalities=data_processors.keys(),
+        )
+        processed_features = []
+        for i in range(sample_num):
+            per_sample_features = apply_data_processor(
+                modality_features=modality_features,
+                data_processors=data_processors,
+                idx=i,
+                is_training=False,
+            )
+            processed_features.append(per_sample_features)
+        collate_fn = get_collate_fn(df_preprocessor=df_preprocessor, data_processors=data_processors)
+        batch = collate_fn(processed_features)
+
+        ret = {}
+        for k in batch:
+            if valid_input and k not in valid_input:
+                continue
+            if onnx_tracing:
+                ret[k] = batch[k].long() if isinstance(batch[k], torch.IntTensor) else batch[k]
+            elif to_numpy:
```

Comment:
If `onnx_tracing ` and `to_numpy ` are both false, it would result in an empty return. Is `to_numpy` necessary?

Reply:
Added the missing condition. If onnx_tracing and to_numpy is false it will output raw tensor without type changes.